test/wasmfs/wasmfs_opfs.c

```diff
@@ -166,6 +166,14 @@ int main(int argc, char* argv[]) {
   assert(strcmp(buf3, msg2) == 0);
   emscripten_console_logf("read message: %s (%d)", buf3, nread);
 
+  // Test reading from an offset.
+  lseek(fd, 1, SEEK_SET);
+  char buf4[100] = {};
+  nread = read(fd, buf4, 4);
+  assert(nread == 4);
+  assert(strcmp(buf4, "ello") == 0);
+  emscripten_console_logf("read message: %s (%d)", buf4, nread);
```

Comment:
Unrelated to this test, but every time I see changes like this I wonder why we have tests like this?  Its this test just a generic file I/O test that should work with any backend not just OPFS?  Same for wasmfs vs legacy FS.   Can't we just have one set of the file I/O tests that work on each of the wasmfs backends and on the legacy FS.

I worry that we have many duplicate and overlapping tests at this point and consolidation is going to be hard (or just never happen).

I know I keep harping on about this but I worry we are moving in the wrong direction.

(@tlively @azakai)

Reply:
Writing new tests was unavoidable at first, but now WasmFS should be feature complete that in principle we could consolidate tests, at least for some things. It's just that it already would be a lot of work :(

Reply:
I agree we should prioritize this now. How about adding no further new tests, and trying to get to this test consolidation before any other significant work on WasmFS?